objects/objects.go

```diff
@@ -13,6 +14,20 @@ func (m Checksum) MarshalJSON() ([]byte, error) {
 	return json.Marshal(fmt.Sprintf("%0x", m[:]))
 }
 
+func (m *Checksum) UnmarshalJSON(data []byte) error {
+	var s string
+	if err := json.Unmarshal(data, &s); err != nil {
+		return err
+	}
+
+	decoded, err := hex.DecodeString(s)
+	if err != nil {
+		return err
+	}
+	copy(m[:], decoded)
```

Comment:
You should check the length of decoded buffer to make sure it is equal to a [32]byte or error out: if decoded is not [32]byte it's not a legal input and we end up with two different situations, either it's < 32 bytes in which cases we have a broken checksum ended with 0s or it's > 32 bytes in which case the input will be truncated.

Reply:
@poolpOrg fixed in  https://github.com/PlakarKorp/plakar/pull/307/commits/cac081352e521e179370df91d2f2e4d6d27de997